### PR TITLE
Upgrade to native Dependabot and auto merge it's PR's

### PR DESCRIPTION
**Pull request by WyriHaximus**
Supersedes / Closes: #186
Modeled after: https://github.com/Roave/BetterReflection/pull/867

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+        run: "composer update --no-interaction --no-progress"
+
+      - name: "Install locked dependencies"
+        if: ${{ matrix.dependencies == 'locked' }}
+        run: "composer install --no-interaction --no-progress"
+
+      - name: "Test Compatibility"
+        run: "vendor/bin/phpunit test/compat"
+
+  demo-scripts:
```
Comment (Ocramius) [inline .github/workflows/continuous-integration.yml]: I smell copy-pasta 😁

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (WyriHaximus) [reply .github/workflows/continuous-integration.yml]: Yes, working on that now 😁

Comment (WyriHaximus) [reply .github/workflows/continuous-integration.yml]: Turned it into a nice pizza 😎

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -0,0 +1,227 @@
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
+
+name: "Continuous Integration"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "5.0.x"
```
Comment (Ocramius) [inline .github/workflows/continuous-integration.yml]: We need to support all branch pushes, IMO

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    name: "Check Coding Standards"
+
+    runs-on: ${{ matrix.operating-system }}
+
+    strategy:
+      matrix:
+        dependencies:
+          - "locked"
+        php-version:
+          - "7.4"
```
Comment (Ocramius) [inline .github/workflows/continuous-integration.yml:183]: Imo enough to run them against 7.4 for now

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .github/workflows/continuous-integration.yml (hunk 3 above)
Comment (WyriHaximus) [reply .github/workflows/continuous-integration.yml:183]: Updated

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (WyriHaximus) [reply .github/workflows/continuous-integration.yml]: 👍

**Review by Ocramius — APPROVED** [review]
LGTM, thanks @WyriHaximus!

**Comment by Ocramius** [comment]
Uh-oh, 8.1 is being annoying again 😬

**Comment by WyriHaximus** [comment]
Want me to not test on `8.1` in this PR? :P

**Comment by Ocramius** [comment]
No, we probably just need to suppres mutants on that code portion, since PHP 8.1 changed behaviour there.

We'd add a TODO for it to get rid of that call.

Also: my own RFC is at fault here 🤣

**Comment by WyriHaximus** [comment]
Do we solve that here or somewhere else?

**Comment by Ocramius** [comment]
Need to do it here: we can't skip testing 8.1, as the MT ia central to this library, after all.

There's inline comments to do that, but I can't find them in the infection docs right now 🤔

**Comment by WyriHaximus** [comment]
Will find it 👍

**Comment by WyriHaximus** [comment]
Ok it's either this or changing `infection.json.dist`

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Awesome, thanks @WyriHaximus!

**Pull request merged**